src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -1104,6 +1104,15 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                         if ($_.Exception -and $_.Exception.WasThrownFromThrowStatement) {
                                             $reason = 'Exception'
                                         }
+                                        # MyCommand can be the script block, so we don't want to show that so check if it's an actual command
+                                        elseif ($myinv.MyCommand -and (Get-Command -Name $myinv.MyCommand -ErrorAction Ignore))
+                                        {
+                                            $reason = $myinv.MyCommand
+                                        }
+                                        # If it's a scriptblock, better to show the command in the scriptblock that had the error
+                                        elseif ($_.CategoryInfo.Activity) {
+                                            $reason = $_.CategoryInfo.Activity
+                                        }
```

Comment:
Maybe move this before line 1108?

Reply:
This ordering is so that if you have:

```powershell
function foo {
  write-error 1
}
```

It will show `foo` as the source of the error and not `write-error`.  In an anonymous scriptblock, it makes sense to show `write-error` as the source rather than the whole scriptblock which is what this fixes.